keepsorted/line_group.go

```diff
@@ -59,17 +59,21 @@ func groupLines(lines []string, metadata blockMetadata) []lineGroup {
 		indents = calculateIndents(lines)
 	}
 
+	countStartDirectives := func(l string) {
+		if strings.Contains(l, metadata.startDirective) {
+			numUnmatchedStartDirectives++
+		} else if strings.Contains(l, metadata.endDirective) {
+			numUnmatchedStartDirectives--
+		}
+	}
+
 	// append a line to both lineRange, and block, if necessary.
 	appendLine := func(i int, l string) {
 		lineRange.append(i)
 		if metadata.opts.Block {
 			block.append(l, metadata.opts)
 		} else if metadata.opts.Group {
-			if strings.Contains(l, metadata.startDirective) {
-				numUnmatchedStartDirectives++
-			} else if strings.Contains(l, metadata.endDirective) {
-				numUnmatchedStartDirectives--
-			}
+			countStartDirectives(l)
```

Comment:
Question/non-blocking: Do you remember why this is for group but not for block mode as well? I'm trying to figure it out and haven't yet. Perhaps you remember.

Reply:
Hah. I had the same thought cross my mind while fixing this, and I was kinda hoping you'd remember

Perhaps because block mode doesn't really need it? The nested keep-sorted lines are always in a situation where there's an unbalanced brace, so this logic has no effect?

It's something that we can probably tweak later if needed